drivers/timer/ti_dmtimer.c

```diff
@@ -0,0 +1,167 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ * Copyright (c) 2024 Texas Instruments Incorporated
+ *	Andrew Davis <afd@ti.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/kernel.h>
+#include <zephyr/spinlock.h>
+
+#include <zephyr/drivers/timer/ti_dmtimer.h>
+
+#define DT_DRV_COMPAT ti_am654_timer
+
+#define TIMER_BASE_ADDR DT_INST_REG_ADDR(0)
+
```

Comment:
@glneo do we need to hardcode the driver to work with only the first/single instance? can we not support multiple instances of the device also? I understand this works well for systick, but can't the timers be used for general purpose use cases as well especially when the device have multiple instances of timer : https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi#n1095

Reply:
Sounds good to me to fix this in future, I believe there is already a method for identifying which timer is systick based on chosen instance. I will recommend adding a TODO noting this that only systick is supported and using it as a general purpose timer is not supported now.